src/metrics/function_error.py

```diff
@@ -23,8 +23,8 @@ def evaluate(self, models: Iterable[ContextModel], num_batches: int = 1) -> Iter
             errs = torch.stack([
                 torch.stack([
                     self.metric.evaluate(
-                        y_batch,
-                        model.evaluate(x_batch, y_batch)
+                        y_batch.to('cpu'),
+                        model.evaluate(x_batch.to('cpu'), y_batch.to('cpu'))
```

Comment:
Moving these to cpu means that we're doing inference on the CPU. This should be avoided at all costs. Instead of changing device here, move the results to cpu only *just* before returning them in each evaluate method.